src/App.tsx

```diff
@@ -1,24 +1,91 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useState, useEffect, useContext } from 'react';
+import classNames from 'classnames';
 
-const USER_ID = 0;
+import { UserWarning } from './components/UserWarning';
+import { Header } from './components/Header';
+import { Main } from './components/Main';
+import { Footer } from './components/Footer';
+
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { PageContext } from './utils/GlobalContext';
 
 export const App: React.FC = () => {
+  const {
+    error,
+    setError,
+    todoList,
+    setTodoList,
+    USER_ID,
+  } = useContext(PageContext);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>(todoList);
```

Comment:
why do you need state for filteredTodos?
in context you alreade has state for all todos.
it is should be enough. just filter todos be active filter where you need it.